src/dmd/statementsem.d

```diff
@@ -4473,6 +4473,94 @@ Statement semanticScope(Statement s, Scope* sc, Statement sbreak, Statement scon
 }
 
 
+/****************************************
+ * If `statement` has code that needs to run in a finally clause
+ * at the end of the current scope, return that code in the form of
+ * a Statement.
+ * Params:
+ *     statement = the statement
+ *     sc = context
+ *     sentry     = set to code executed upon entry to the scope
+ *     sexception = set to code executed upon exit from the scope via exception
+ *     sfinally   = set to code executed in finally block
+ * Returns:
+ *    code to be run in the finally clause
+ */
+Statement scopeCode(Statement statement, Scope* sc, Statement* sentry, Statement* sexception, Statement* sfinally)
+{
+    *sentry = null;
+    *sexception = null;
+    *sfinally = null;
```

Comment:
You could turn these into `out` parameters.

Reply:
Preferably in a follow-up, however. As that would alter the callers too.

Reply:
Good call.  Can I do it in a new PR since I think will change a bunch of callsites?

Reply:
Please resist the urge to do anything more than move the code in this PR. Fixing other things is for separate PRs.